LaTeX/LaTeX.sublime-syntax

```diff
@@ -110,66 +122,128 @@ contexts:
 
   macros:
     - include: scope:text.tex#macros
-    - match: |-
-        (?x)
-        (
-          (\\)
-          (?:(?:new|renew|provide)command\*?)
-        )
-        (?:
-          (\{)(\\[A-Za-z@]+)(\})
-          | (\\[A-Za-z@])+
-        )
-        (?:(\[)(?:[^\]]*)(\]))*
-        (\{)
+    - match: (\\)(?:new|renew|provide)command\*?
+      scope: keyword.declaration.function.latex storage.modifier.newcommand.latex
       captures:
-        1: support.function.newcommand.latex storage.modifier.newcommand.latex
-        2: punctuation.definition.backslash.latex
-        3: punctuation.definition.group.brace.begin.latex
-        4: support.function.latex entity.name.newcommand.latex
-        5: punctuation.definition.group.brace.end.latex
-        6: support.function.latex
-        7: punctuation.definition.group.bracket.begin.latex
-        8: punctuation.definition.group.bracket.end.latex
-        9: punctuation.definition.group.brace.begin.latex
+        1: punctuation.definition.backslash.latex
       push:
-        - meta_scope: meta.function.newcommand.latex
-        - match: '\}'
-          scope: punctuation.definition.group.brace.end.latex
-          pop: true
+        - newcommand-definition
+        - newcommand-optarg
+        - newcommand-argspec
+        - newcommand-commandname
+    - match: (\\)DeclareMathOperator\*?
+      scope: support.function.declare-math-operator.latex storage.modifier.newcommand.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      push:
+        - newcommand-definition
+        - newcommand-commandname
+
+  newcommand-argspec:
+    - match: \[
+      scope: punctuation.definition.group.bracket.begin.latex
+      push:
+        - clear_scopes: 1
+        - meta_scope: meta.function.parameters.latex
```

Comment:
What we probably should also try to do is to avoid anonymous contexts. 

The 2 major reasons are:

1. support inheritance: only named contexts can be addressed in case a 3rd-party syntax wants to extend one of the contexts.
2. diagnostics: ST can display the context which is involved in highlighting via `ctrl+shift+alt+p`. Anonymous contexts are hard to find there.


_Note: I haven't made that change in my probosal as I wanted to concentrate on meta the scope._

Reply:
With this change, the `newcommand-block-body` is essentially the same as the anonymous context pushed in `macros` from `TeX.sublime-syntax`, except for the scope names. Could we unify that in a sensible way?